x/archivedb/db.go

```diff
@@ -121,64 +120,11 @@ func (db *archiveDB) Get(key []byte, height uint64) ([]byte, uint64, error) {
 }
 
 // Creates a new batch to append database changes in a given height
-func (db *archiveDB) NewBatch() (batchWithHeight, error) {
-	var nextHeightBytes [8]byte
-
+func (db *archiveDB) NewBatch(height uint64) (dbBatchWithHeight, error) {
```

Comment:
I think it is more consistent with user expectation that this would return `*dbBatchWithHeight`